ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Node.hs

```diff
@@ -79,6 +85,33 @@ data TPraosLeaderCredentials c = TPraosLeaderCredentials {
   , tpraosLeaderCredentialsIsCoreNode :: TPraosIsCoreNode c
   }
 
+-- | Genesis Shelley staking configuration.
+--
+-- This allows us to configure some initial stake pools and delegation to them,
+-- in order to test Praos in a static configuration, without requiring on-chain
+-- registration and delegation.
+--
+-- For simplicity, pools defined in the genesis staking do not pay deposits for
+-- their registration.
+data ShelleyGenesisStaking c = ShelleyGenesisStaking {
+    -- | Pools to register
+    sgsPools :: !(Map (SL.KeyHash c) (SL.PoolParams c))
```

Comment:
For my understanding: what exactly is the `SL.KeyHash c` here? A hash of the public key of the *pool*?

Reply:
Exactly. Which doesn't correspond to anything else in particular; it could be a totally fresh key, but in the testing I set it to the payment key of the delegate in order to allow the same nodes to issue blocks under each protocol.

Actually, there's a bug in c-l-s test generators which assume this to be a _staking_ key. This causes occasional failures in these tests when we generate a pool retirement cert. I've updated c-l-s to fix this (https://github.com/input-output-hk/cardano-ledger-specs/pull/1393) and will update this PR accordingly